src/weather_gen.h

```diff
@@ -3,13 +3,82 @@
 #define CATA_SRC_WEATHER_GEN_H
 
 #include <string>
+#include <climits>
 
 #include "calendar.h"
+#include "color.h"
 
 struct tripoint;
 class JsonObject;
+using weather_type = int;
```

Comment:
Rather than using a typedef for an `int`, if you move to generic factory it would probably be clearer to have a `weather_type_id` based on `string_id` or `int_id`.